oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/RelatedSelectField/ExternalApiModal.jsx

```diff
@@ -0,0 +1,230 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { i18next } from "@translations/oarepo_ui/i18next";
+import { Modal, Button, Grid, Header, Segment, Icon } from "semantic-ui-react";
+import { OverridableContext } from "react-overridable";
+import {
+  EmptyResults,
+  Error,
+  ReactSearchKit,
+  ResultsLoader,
+  SearchBar,
+  InvenioSearchApi,
+  Pagination,
+  ResultsPerPage,
+} from "react-searchkit";
+import { ExternalApiResultsList } from "./ExternalApiResultsList";
+import { useFormikContext } from "formik";
+import _isEmpty from "lodash/isEmpty";
+
+// have the same component in another module. Where could we place such
+// small components to be reused
+const resultsPerPageLabel = (cmp) => (
+  <React.Fragment>
+    {cmp} {i18next.t("resultsPerPage")}
+  </React.Fragment>
+);
+
+//  providing custom empty error message because we are initializing app with a query string and the button to reset search effectivelly does nothing (it is resetting to already existing query string so I thought it is useless)
+// and I dont see a reasonable way to override resetQuery action from react searchkit
+export const EmptyResultsElement = ({ queryString }) => {
+  return (
+    <Segment placeholder textAlign="center">
+      <Header icon>
+        <Icon name="search" />
+      </Header>
+      {queryString && (
+        <em>
+          {i18next.t("We couldn't find any matches for ")} "{queryString}"
+        </em>
+      )}
+      <br />
+    </Segment>
+  );
+};
+
+EmptyResultsElement.propTypes = {
+  queryString: PropTypes.string,
+  resetQuery: PropTypes.func.isRequired,
+  extraContent: PropTypes.node,
+};
+
+const overriddenComponents = {
+  ["EmptyResults.element"]: EmptyResultsElement,
+};
+
+export const ExternalApiModal = ({
+  searchConfig,
+  open,
+  onClose,
+  handleAddingExternalApiSuggestion,
+  fieldPath,
+  serializeSuggestions,
+}) => {
+  const [externalApiRecord, setExternalApiRecord] = useState({});
+  const { setFieldValue } = useFormikContext();
+  const searchApi = new InvenioSearchApi(searchConfig.searchApi);
+  const handleExternalRecordChange = (record) => {
+    setExternalApiRecord(record);
+  };
+  return (
+    <Modal open={open} onClose={onClose}>
+      <Modal.Header as="h6" className="pt-10 pb-10">
+        <Grid>
+          <Grid.Column floated="left">
+            <Header as="h2">Search results from external API</Header>
```

Comment:
This should be a customizable prop (defined by user of the `RelatedSelectField`)